homeassistant/components/vodafone_station/coordinator.py

```diff
@@ -0,0 +1,164 @@
+        return self._name
+
+    @property
+    def ip_address(self) -> str | None:
+        """Get IP address."""
+        return self._ip_address
+
+    @property
+    def last_activity(self) -> datetime | None:
+        """Return device last activity."""
+        return self._last_activity
+
+    @property
+    def wifi(self) -> str | None:
+        """Return device WIFi connection."""
+        return self._wifi
+
+
+class VodafoneStationRouter(DataUpdateCoordinator):
+    """Queries router running Vodafone Station firmware."""
+
+    def __init__(
+        self, host: str, ssl: bool, username: str, password: str, hass: HomeAssistant
+    ) -> None:
+        """Initialize the scanner."""
+
+        self._host = host
+        self._devices: dict[str, VodafoneStationDeviceInfo] = {}
+        self._data: dict[str, str] = {}
+        self._on_close: list[Callable] = []
+
+        self.api = VodafoneStationApi(host, ssl, username, password)
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=f"{DOMAIN}-{host}-coordinator",
+            update_interval=timedelta(seconds=30),
+        )
+
+    @callback
+    def async_listener(self) -> None:
+        """Fetch data from the router."""
+
+        asyncio.run_coroutine_threadsafe(self._async_update_data(), self.hass.loop)
+
+    @callback
+    def async_on_close(self, func: CALLBACK_TYPE) -> None:
+        """Add a function to call when router is closed."""
+        self._on_close.append(func)
+
+    async def _async_update_data(self) -> bool:
+        """Update router data."""
+        _LOGGER.debug("Polling Vodafone Station host: %s", self._host)
+        try:
+            logged = await self.api.login()
+        except (asyncio.exceptions.TimeoutError, aiohttp.ClientConnectorError) as err:
+            _LOGGER.warning("Connection error for %s", self._host)
+            raise UpdateFailed(f"Error fetching data: {repr(err)}") from err
+
+        if not logged:
+            raise ConfigEntryAuthFailed
+
+        devices = await self.api.get_all_devices()
+        dev_info: VodafoneStationDevice
+        for _, dev_info in devices.items():
+            dev = VodafoneStationDeviceInfo(dev_info.mac, dev_info.name)
+            dev.update(dev_info)
+            self._devices[dev_info.mac] = dev
+        self._data = await self.api.get_user_data()
+
+        await self.api.logout()
+
+        return True
+
+    @property
+    def devices(self) -> dict[str, VodafoneStationDeviceInfo]:
+        """Return a list of devices."""
+        return self._devices
+
+    @property
+    def signal_device_new(self) -> str:
+        """Event specific per Vodafone Station entry to signal new device."""
+        return f"{DOMAIN}-device-new-{self._host}"
+
+    @property
+    def signal_device_update(self) -> str:
+        """Event specific per Vodafone Station entry to signal updates in devices."""
+        return f"{DOMAIN}-device-update-{self._host}"
```

Comment:
The signal should be unique and which means it should use the mac

Reply:
Ok, will review.
There are at least other 2 integrations that need to be fixed:
- `freebox`
- `asuswrt`

and maybe even `icloud`; at minimum the comment is wrong ;-)

Simone
-

Reply:
Done